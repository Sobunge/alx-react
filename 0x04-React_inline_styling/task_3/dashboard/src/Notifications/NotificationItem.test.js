import React from 'react';
import NotificationItem from './NotificationItem';
import { shallow } from 'enzyme';

describe('rendering components', () => {
  it('renders NotificationItem component without crashing', () => {
    const wrapper = shallow(<NotificationItem />);
    expect(wrapper.exists()).toBe(true);
  });

  it('renders correct html from type="default" value="test" props', () => {
    const wrapper = shallow(<NotificationItem type="default" value="test" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correct html from html="<u>test</u>" props', () => {
    const wrapper = shallow(<NotificationItem html="<u>test</u>" />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe('onclick event behaves as it should', () => {
  it('should call markAsRead with correct id', () => {
    const spy = jest.fn();
    const wrapper = shallow(<NotificationItem value="test item" markAsRead={spy} id={1} />);
    wrapper.find('li').simulate('click');
    expect(spy).toBeCalledTimes(1);
    expect(spy).toBeCalledWith(1);
  });
});
