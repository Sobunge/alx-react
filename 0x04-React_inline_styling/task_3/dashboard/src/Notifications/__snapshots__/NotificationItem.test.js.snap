// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering components renders correct html from html="<u>test</u>" props 1`] = `"<li data-urgent=\\"true\\" class=\\"urgent_cyonix\\"><u>test</u></li>"`;

exports[`rendering components renders correct html from type="default" value="test" props 1`] = `"<li class=\\"default_2c02es\\" data-notification-type=\\"default\\">test</li>"`;
